src/stream/src/executor_v2/error.rs

```diff
@@ -47,6 +49,9 @@ pub enum StreamExecutorError {
 
     #[error("Channel `{0}` closed")]
     ChannelClosed(String),
+
+    #[error("Failed to align barrier: expected {0:?} but got {1:?}")]
+    AlignBarrier(Box<Barrier>, Box<Barrier>),
```

Comment:
Could you please help to use this error in `Merge`?

https://github.com/singularity-data/risingwave/blob/9499dc7434d884961ef178115470a104ef24415e/src/stream/src/executor_v2/merge.rs#L182-L187

Reply:
Next time for sure 🤪

Reply:
And I think it's better to use two different errors at two different places.

Reply:
They're both for aligning barriers. 😭